test/integration/initial-dns-seedlist-discovery/dns_seedlist.test.ts

```diff
@@ -0,0 +1,162 @@
+  });
+
+  afterEach(async function () {
+    sinon.restore();
+    await client.close();
+  });
+
+  const restoreDNS =
+    api =>
+    async (...args) => {
+      sinon.restore();
+      return await dns.promises[api](...args);
+    };
+
+  describe('when SRV record look up times out', () => {
+    beforeEach(() => {
+      sinon
+        .stub(dns.promises, 'resolveSrv')
+        .onFirstCall()
+        .rejects(new DNSTimeoutError())
+        .onSecondCall()
+        .callsFake(restoreDNS('resolveSrv'));
+    });
+
+    afterEach(async function () {
+      sinon.restore();
+    });
+
+    it('retries timeout error', metadata, async () => {
+      await client.connect();
+    });
+  });
+
+  describe('when TXT record look up times out', () => {
+    beforeEach(() => {
+      sinon
+        .stub(dns.promises, 'resolveTxt')
+        .onFirstCall()
+        .rejects(new DNSTimeoutError())
+        .onSecondCall()
+        .callsFake(restoreDNS('resolveTxt'));
+    });
+
+    afterEach(async function () {
+      sinon.restore();
+    });
+
+    it('retries timeout error', metadata, async () => {
+      await client.connect();
+    });
+  });
+
+  describe('when SRV record look up times out twice', () => {
+    beforeEach(() => {
+      sinon
+        .stub(dns.promises, 'resolveSrv')
+        .onFirstCall()
+        .rejects(new DNSTimeoutError())
+        .onSecondCall()
+        .rejects(new DNSTimeoutError())
+        .onThirdCall()
+        .callsFake(restoreDNS('resolveSrv'));
+    });
+
+    afterEach(async function () {
+      sinon.restore();
+    });
+
+    it('throws timeout error', metadata, async () => {
+      const error = await client.connect().catch(error => error);
+      expect(error).to.be.instanceOf(DNSTimeoutError);
+    });
+  });
+
+  describe('when TXT record look up times out twice', () => {
+    beforeEach(() => {
+      sinon
+        .stub(dns.promises, 'resolveTxt')
+        .onFirstCall()
+        .rejects(new DNSTimeoutError())
+        .onSecondCall()
+        .rejects(new DNSTimeoutError())
+        .onThirdCall()
+        .callsFake(restoreDNS('resolveTxt'));
+    });
+
+    afterEach(async function () {
+      sinon.restore();
+    });
+
+    it('throws timeout error', metadata, async () => {
+      const error = await client.connect().catch(error => error);
+      expect(error).to.be.instanceOf(DNSTimeoutError);
+    });
+  });
+
+  describe('when SRV record look up throws a non-timeout error', () => {
+    beforeEach(() => {
+      sinon
+        .stub(dns.promises, 'resolveSrv')
+        .onFirstCall()
+        .rejects(new DNSSomethingError())
+        .onSecondCall()
+        .callsFake(restoreDNS('resolveSrv'));
+    });
+
+    afterEach(async function () {
+      sinon.restore();
+    });
+
+    it('throws that error', metadata, async () => {
+      const error = await client.connect().catch(error => error);
```

Comment:
should we assert that we didn't retry here too?

Reply:
done